### PR TITLE
Add new and improve existing `app-store-connect` actions

### PR DESCRIPTION
**Pull request by priitlatt**
This PR adds new features and some bugfixes/improvements to `app-store-connect` tool.

**Highlights**

Add new CLI actions to fetch information about App Store versions, App Store review submissions and review submission items.

Fix creating App Store review submission items and provisioning profiles that are not ad-hoc or development type.

Show more descriptive error message when creating App Store review submission item fails because required values are missing for application default locale on respective App Store Version.

**New actions:**
- `app-store-connect app-store-versions get`,
- `app-store-connect review-submissions items`,
- `app-store-connect review-submission-items delete`.

**Updated actions:**
- `app-store-connect app-store-versions localizations`,
- `app-store-connect review-submission-items create`,
- `app-store-connect create-profile`.

### CODE REVIEW

**Comment by priitlatt** [comment]
Preview of improved error message for `app-store-connect review-submission-items create` if creating review submission item fails due to missing values on App Store version localization:

Screenshot 2023-12-04 at 13 17 14

File: src/codemagic/apple/app_store_connect/versioning/review_submissions.py [hunk 1]
```diff
@@ -99,3 +100,11 @@ def modify(
             json=payload,
         ).json()
         return ReviewSubmission(response["data"])
+
+    def list_items(self, review_submission: Union[LinkedResourceData, ResourceId]):
```
Comment (artemii-yanushevskyi) [inline src/codemagic/apple/app_store_connect/versioning/review_submissions.py:104]: Return type?

Comment (priitlatt) [reply src/codemagic/apple/app_store_connect/versioning/review_submissions.py:104]: Yes! Somehow I forgot about it.

Comment (priitlatt) [reply src/codemagic/apple/app_store_connect/versioning/review_submissions.py]: Added in [90dd967](https://github.com/codemagic-ci-cd/cli-tools/pull/380/commits/90dd967934421ab7cc02fb6d7853127ac02f70ca).

**Pull request merged**